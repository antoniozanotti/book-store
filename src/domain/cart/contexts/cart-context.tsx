"use client";

import {
  createContext,
  useContext,
  useReducer,
  ReactNode,
  useEffect,
} from "react";
import { CartType } from "../types/cart-type";
import { CartContextType } from "../types/cart-context-type";
import { useLocalStorage } from "usehooks-ts";
import { useProductsByIdsQuery } from "@/domain/product/queries/use-products-by-ids-query";
import { cartReducer } from "../reducers/cart-reducer";

const initialState: CartType = { items: [] };

const CartContext = createContext<CartContextType>({
  state: initialState,
  dispatch: () => {},
  getCartItemsQty: () => {},
  cartItemsSubtotal: () => {},
});

export function CartProvider({ children }: { children: ReactNode }) {
  const [localState, setLocalState] = useLocalStorage<CartType>(
    "cart",
    initialState
  );
  const [state, dispatch] = useReducer(cartReducer, localState);
  const cartItems = state.items;
  const responses = useProductsByIdsQuery(cartItems.map((item) => item.id));

  function getCartItemsQty(): number {
    const cartItems = state.items;
    return cartItems.reduce((acc, cur) => {
      return acc + cur.quantity;
    }, 0);
  }

  function cartItemsSubtotal(): number {
    let subTotal = 0;
    cartItems.forEach((item) => {
      responses.some((response) => {
        const product = response.data;
        if (!product) {
          return true;
        }
        if (item.id == product.id) {
          subTotal += parseInt(product.price.toString()) * item.quantity;
          return true;
        }
      });
    });
    return subTotal;
  }

  useEffect(() => {
    setLocalState(state);
  }, [state, setLocalState]);

  return (
    <CartContext.Provider
      value={{ state, dispatch, getCartItemsQty, cartItemsSubtotal }}
    >
      {children}
    </CartContext.Provider>
  );
}

export function useCartContext(): CartContextType {
  return useContext(CartContext);
}
