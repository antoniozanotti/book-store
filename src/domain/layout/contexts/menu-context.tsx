"use client";

import { createContext, useContext, useReducer, ReactNode } from "react";
import { MenuType } from "../types/menu-type";
import { MenuContextType } from "../types/menu-context-type";
import { menuReducer } from "../reducers/menu-reducer";

const initialState: MenuType = { isOpen: false };

const MenuContext = createContext<MenuContextType>({
  state: initialState,
  dispatch: () => {},
});

export function MenuProvider({ children }: { children: ReactNode }) {
  const [state, dispatch] = useReducer(menuReducer, initialState);

  return (
    <MenuContext.Provider value={{ state, dispatch }}>
      {children}
    </MenuContext.Provider>
  );
}

export function useMenuContext(): MenuContextType {
  return useContext(MenuContext);
}
