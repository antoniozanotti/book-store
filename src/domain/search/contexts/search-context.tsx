"use client";

import { createContext, useContext, useReducer, ReactNode } from "react";
import { SearchContextType } from "../types/search-context-type";
import { useSearchParams } from "next/navigation";
import { getSearchParams } from "../services/search-service";
import { searchReducer } from "../reducers/search-reducer";

const SearchContext = createContext<SearchContextType>({} as SearchContextType);

export function SearchProvider({ children }: { children: ReactNode }) {
  const searchParams = useSearchParams();
  const { keyword, categories, minPrice, maxPrice, minYear, maxYear, orderBy } =
    getSearchParams(searchParams);
  const initialStateBySearchParams = {
    keyword: keyword,
    categories: categories,
    minprice: minPrice,
    maxprice: maxPrice,
    minyear: minYear,
    maxyear: maxYear,
    orderby: orderBy,
  };
  const [state, dispatch] = useReducer(
    searchReducer,
    initialStateBySearchParams
  );

  return (
    <SearchContext.Provider value={{ state, dispatch }}>
      {children}
    </SearchContext.Provider>
  );
}

export function useSearchContext(): SearchContextType {
  return useContext(SearchContext);
}
