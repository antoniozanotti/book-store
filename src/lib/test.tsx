import React, { ReactElement, ReactNode } from "react";
import {
  render,
  RenderOptions,
  RenderResult,
  Screen,
} from "@testing-library/react";

function AllTheProviders({ children }: { children: React.ReactNode }) {
  return <div data-testid="container">{children}</div>;
}

function customRender(
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderResult {
  return render(ui, { wrapper: AllTheProviders, ...options });
}

function firstComponent(screen: Screen) {
  return screen.getByTestId("container").firstChild;
}

export * from "@testing-library/react";
export * from "@testing-library/user-event";
export { customRender as render, firstComponent };
import "@testing-library/jest-dom";
import { usePathname } from "next/navigation";

jest.mock("next/navigation", () => {
  return {
    useRouter() {
      return { push: () => {} };
    },
    useParams: () => ({}),
    notFound: () => null,
    usePathname: () => null,
    useSearchParams: () => null,
  };
});
