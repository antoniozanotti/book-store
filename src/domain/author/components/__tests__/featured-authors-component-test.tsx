import { render } from "@/lib/test";
import { FeaturedAuthorsComponent } from "../featured-authors-component";
import { FeaturedAuthorsProvider } from "@/domain/author/contexts/featured-authors-context";
import { authorsWithPageMock } from "../../__tests__/authors-with-page-mock";

function featuredAuthorsComponentFactory(){
  return render(<FeaturedAuthorsProvider authors={authorsWithPageMock}><FeaturedAuthorsComponent /></FeaturedAuthorsProvider>);
}

describe("FeaturedAuthorsComponent", () => {
  it("should render", async () => {
    const { container } = featuredAuthorsComponentFactory();
    expect(container).toMatchSnapshot();
  });
});
