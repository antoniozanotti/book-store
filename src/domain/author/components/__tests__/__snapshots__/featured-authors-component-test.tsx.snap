// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedAuthorsComponent should render 1`] = `
<div>
  <div
    data-testid="container"
  >
    <section
      class="p-4 pb-10 md:p-10"
    >
      <h2
        class="font-primary text-pretty text-4xl sm:text-5xl md:text-6xl lg:text-7xl"
      >
        Featured Authors
      </h2>
      <div
        class="snap-x overflow-x-auto flex sm:flex-wrap sm:justify-center sm:gap-2 text-center"
      >
        <article
          class="p-5 w-[227px] h-[267px] flex-none text-sm text-pretty"
        >
          <a
            href="/author/kevin-kwan"
            rel="author"
            title="Kevin Kwan"
          >
            <figure
              class="bg-green-300 w-[187px] h-[187px] relative overflow-hidden rounded-full"
            >
              <img
                alt="Kevin Kwan"
                class="transition-all w-[187px] h-[187px] hover:w-[207px] hover:h-[207px] hover:-mt-[10px] object-cover"
                data-nimg="1"
                decoding="async"
                height="207"
                loading="lazy"
                sizes="207px"
                src="https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_3840,h_3840,g_auto/f_auto/q_auto/v1/author/kevin-kwan?_a=BAVARSDW0"
                srcset="https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_16,h_16,g_auto/f_auto/q_auto/v1/author/kevin-kwan?_a=BAVARSDW0 16w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_32,h_32,g_auto/f_auto/q_auto/v1/author/kevin-kwan?_a=BAVARSDW0 32w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_48,h_48,g_auto/f_auto/q_auto/v1/author/kevin-kwan?_a=BAVARSDW0 48w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_64,h_64,g_auto/f_auto/q_auto/v1/author/kevin-kwan?_a=BAVARSDW0 64w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_96,h_96,g_auto/f_auto/q_auto/v1/author/kevin-kwan?_a=BAVARSDW0 96w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_128,h_128,g_auto/f_auto/q_auto/v1/author/kevin-kwan?_a=BAVARSDW0 128w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_256,h_256,g_auto/f_auto/q_auto/v1/author/kevin-kwan?_a=BAVARSDW0 256w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_384,h_384,g_auto/f_auto/q_auto/v1/author/kevin-kwan?_a=BAVARSDW0 384w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_640,h_640,g_auto/f_auto/q_auto/v1/author/kevin-kwan?_a=BAVARSDW0 640w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_750,h_750,g_auto/f_auto/q_auto/v1/author/kevin-kwan?_a=BAVARSDW0 750w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_828,h_828,g_auto/f_auto/q_auto/v1/author/kevin-kwan?_a=BAVARSDW0 828w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_1080,h_1080,g_auto/f_auto/q_auto/v1/author/kevin-kwan?_a=BAVARSDW0 1080w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_1200,h_1200,g_auto/f_auto/q_auto/v1/author/kevin-kwan?_a=BAVARSDW0 1200w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_1920,h_1920,g_auto/f_auto/q_auto/v1/author/kevin-kwan?_a=BAVARSDW0 1920w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_2048,h_2048,g_auto/f_auto/q_auto/v1/author/kevin-kwan?_a=BAVARSDW0 2048w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_3840,h_3840,g_auto/f_auto/q_auto/v1/author/kevin-kwan?_a=BAVARSDW0 3840w"
                style="color: transparent;"
                width="207"
              />
            </figure>
          </a>
          <p
            class="font-primary text-pretty text-2xl line-clamp-2 mt-2"
          >
            <a
              href="/author/kevin-kwan"
              rel="author"
              title="Kevin Kwan"
            >
              Kevin Kwan
            </a>
          </p>
        </article>
        <article
          class="p-5 w-[227px] h-[267px] flex-none text-sm text-pretty"
        >
          <a
            href="/author/julia-cameron"
            rel="author"
            title="Julia Cameron"
          >
            <figure
              class="bg-green-300 w-[187px] h-[187px] relative overflow-hidden rounded-full"
            >
              <img
                alt="Julia Cameron"
                class="transition-all w-[187px] h-[187px] hover:w-[207px] hover:h-[207px] hover:-mt-[10px] object-cover"
                data-nimg="1"
                decoding="async"
                height="207"
                loading="lazy"
                sizes="207px"
                src="https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_3840,h_3840,g_auto/f_auto/q_auto/v1/author/julia-cameron?_a=BAVARSDW0"
                srcset="https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_16,h_16,g_auto/f_auto/q_auto/v1/author/julia-cameron?_a=BAVARSDW0 16w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_32,h_32,g_auto/f_auto/q_auto/v1/author/julia-cameron?_a=BAVARSDW0 32w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_48,h_48,g_auto/f_auto/q_auto/v1/author/julia-cameron?_a=BAVARSDW0 48w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_64,h_64,g_auto/f_auto/q_auto/v1/author/julia-cameron?_a=BAVARSDW0 64w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_96,h_96,g_auto/f_auto/q_auto/v1/author/julia-cameron?_a=BAVARSDW0 96w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_128,h_128,g_auto/f_auto/q_auto/v1/author/julia-cameron?_a=BAVARSDW0 128w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_256,h_256,g_auto/f_auto/q_auto/v1/author/julia-cameron?_a=BAVARSDW0 256w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_384,h_384,g_auto/f_auto/q_auto/v1/author/julia-cameron?_a=BAVARSDW0 384w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_640,h_640,g_auto/f_auto/q_auto/v1/author/julia-cameron?_a=BAVARSDW0 640w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_750,h_750,g_auto/f_auto/q_auto/v1/author/julia-cameron?_a=BAVARSDW0 750w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_828,h_828,g_auto/f_auto/q_auto/v1/author/julia-cameron?_a=BAVARSDW0 828w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_1080,h_1080,g_auto/f_auto/q_auto/v1/author/julia-cameron?_a=BAVARSDW0 1080w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_1200,h_1200,g_auto/f_auto/q_auto/v1/author/julia-cameron?_a=BAVARSDW0 1200w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_1920,h_1920,g_auto/f_auto/q_auto/v1/author/julia-cameron?_a=BAVARSDW0 1920w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_2048,h_2048,g_auto/f_auto/q_auto/v1/author/julia-cameron?_a=BAVARSDW0 2048w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_3840,h_3840,g_auto/f_auto/q_auto/v1/author/julia-cameron?_a=BAVARSDW0 3840w"
                style="color: transparent;"
                width="207"
              />
            </figure>
          </a>
          <p
            class="font-primary text-pretty text-2xl line-clamp-2 mt-2"
          >
            <a
              href="/author/julia-cameron"
              rel="author"
              title="Julia Cameron"
            >
              Julia Cameron
            </a>
          </p>
        </article>
        <article
          class="p-5 w-[227px] h-[267px] flex-none text-sm text-pretty"
        >
          <a
            href="/author/david-byrne"
            rel="author"
            title="David Byrne"
          >
            <figure
              class="bg-green-300 w-[187px] h-[187px] relative overflow-hidden rounded-full"
            >
              <img
                alt="David Byrne"
                class="transition-all w-[187px] h-[187px] hover:w-[207px] hover:h-[207px] hover:-mt-[10px] object-cover"
                data-nimg="1"
                decoding="async"
                height="207"
                loading="lazy"
                sizes="207px"
                src="https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_3840,h_3840,g_auto/f_auto/q_auto/v1/author/david-byrne?_a=BAVARSDW0"
                srcset="https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_16,h_16,g_auto/f_auto/q_auto/v1/author/david-byrne?_a=BAVARSDW0 16w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_32,h_32,g_auto/f_auto/q_auto/v1/author/david-byrne?_a=BAVARSDW0 32w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_48,h_48,g_auto/f_auto/q_auto/v1/author/david-byrne?_a=BAVARSDW0 48w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_64,h_64,g_auto/f_auto/q_auto/v1/author/david-byrne?_a=BAVARSDW0 64w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_96,h_96,g_auto/f_auto/q_auto/v1/author/david-byrne?_a=BAVARSDW0 96w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_128,h_128,g_auto/f_auto/q_auto/v1/author/david-byrne?_a=BAVARSDW0 128w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_256,h_256,g_auto/f_auto/q_auto/v1/author/david-byrne?_a=BAVARSDW0 256w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_384,h_384,g_auto/f_auto/q_auto/v1/author/david-byrne?_a=BAVARSDW0 384w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_640,h_640,g_auto/f_auto/q_auto/v1/author/david-byrne?_a=BAVARSDW0 640w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_750,h_750,g_auto/f_auto/q_auto/v1/author/david-byrne?_a=BAVARSDW0 750w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_828,h_828,g_auto/f_auto/q_auto/v1/author/david-byrne?_a=BAVARSDW0 828w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_1080,h_1080,g_auto/f_auto/q_auto/v1/author/david-byrne?_a=BAVARSDW0 1080w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_1200,h_1200,g_auto/f_auto/q_auto/v1/author/david-byrne?_a=BAVARSDW0 1200w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_1920,h_1920,g_auto/f_auto/q_auto/v1/author/david-byrne?_a=BAVARSDW0 1920w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_2048,h_2048,g_auto/f_auto/q_auto/v1/author/david-byrne?_a=BAVARSDW0 2048w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_3840,h_3840,g_auto/f_auto/q_auto/v1/author/david-byrne?_a=BAVARSDW0 3840w"
                style="color: transparent;"
                width="207"
              />
            </figure>
          </a>
          <p
            class="font-primary text-pretty text-2xl line-clamp-2 mt-2"
          >
            <a
              href="/author/david-byrne"
              rel="author"
              title="David Byrne"
            >
              David Byrne
            </a>
          </p>
        </article>
        <article
          class="p-5 w-[227px] h-[267px] flex-none text-sm text-pretty"
        >
          <a
            href="/author/donna-tartt"
            rel="author"
            title="Donna Tartt"
          >
            <figure
              class="bg-green-300 w-[187px] h-[187px] relative overflow-hidden rounded-full"
            >
              <img
                alt="Donna Tartt"
                class="transition-all w-[187px] h-[187px] hover:w-[207px] hover:h-[207px] hover:-mt-[10px] object-cover"
                data-nimg="1"
                decoding="async"
                height="207"
                loading="lazy"
                sizes="207px"
                src="https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_3840,h_3840,g_auto/f_auto/q_auto/v1/author/donna-tartt?_a=BAVARSDW0"
                srcset="https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_16,h_16,g_auto/f_auto/q_auto/v1/author/donna-tartt?_a=BAVARSDW0 16w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_32,h_32,g_auto/f_auto/q_auto/v1/author/donna-tartt?_a=BAVARSDW0 32w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_48,h_48,g_auto/f_auto/q_auto/v1/author/donna-tartt?_a=BAVARSDW0 48w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_64,h_64,g_auto/f_auto/q_auto/v1/author/donna-tartt?_a=BAVARSDW0 64w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_96,h_96,g_auto/f_auto/q_auto/v1/author/donna-tartt?_a=BAVARSDW0 96w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_128,h_128,g_auto/f_auto/q_auto/v1/author/donna-tartt?_a=BAVARSDW0 128w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_256,h_256,g_auto/f_auto/q_auto/v1/author/donna-tartt?_a=BAVARSDW0 256w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_384,h_384,g_auto/f_auto/q_auto/v1/author/donna-tartt?_a=BAVARSDW0 384w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_640,h_640,g_auto/f_auto/q_auto/v1/author/donna-tartt?_a=BAVARSDW0 640w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_750,h_750,g_auto/f_auto/q_auto/v1/author/donna-tartt?_a=BAVARSDW0 750w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_828,h_828,g_auto/f_auto/q_auto/v1/author/donna-tartt?_a=BAVARSDW0 828w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_1080,h_1080,g_auto/f_auto/q_auto/v1/author/donna-tartt?_a=BAVARSDW0 1080w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_1200,h_1200,g_auto/f_auto/q_auto/v1/author/donna-tartt?_a=BAVARSDW0 1200w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_1920,h_1920,g_auto/f_auto/q_auto/v1/author/donna-tartt?_a=BAVARSDW0 1920w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_2048,h_2048,g_auto/f_auto/q_auto/v1/author/donna-tartt?_a=BAVARSDW0 2048w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_3840,h_3840,g_auto/f_auto/q_auto/v1/author/donna-tartt?_a=BAVARSDW0 3840w"
                style="color: transparent;"
                width="207"
              />
            </figure>
          </a>
          <p
            class="font-primary text-pretty text-2xl line-clamp-2 mt-2"
          >
            <a
              href="/author/donna-tartt"
              rel="author"
              title="Donna Tartt"
            >
              Donna Tartt
            </a>
          </p>
        </article>
        <article
          class="p-5 w-[227px] h-[267px] flex-none text-sm text-pretty"
        >
          <a
            href="/author/jeremy-mccarter"
            rel="author"
            title="Jeremy McCarter"
          >
            <figure
              class="bg-green-300 w-[187px] h-[187px] relative overflow-hidden rounded-full"
            >
              <img
                alt="Jeremy McCarter"
                class="transition-all w-[187px] h-[187px] hover:w-[207px] hover:h-[207px] hover:-mt-[10px] object-cover"
                data-nimg="1"
                decoding="async"
                height="207"
                loading="lazy"
                sizes="207px"
                src="https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_3840,h_3840,g_auto/f_auto/q_auto/v1/author/jeremy-mccarter?_a=BAVARSDW0"
                srcset="https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_16,h_16,g_auto/f_auto/q_auto/v1/author/jeremy-mccarter?_a=BAVARSDW0 16w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_32,h_32,g_auto/f_auto/q_auto/v1/author/jeremy-mccarter?_a=BAVARSDW0 32w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_48,h_48,g_auto/f_auto/q_auto/v1/author/jeremy-mccarter?_a=BAVARSDW0 48w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_64,h_64,g_auto/f_auto/q_auto/v1/author/jeremy-mccarter?_a=BAVARSDW0 64w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_96,h_96,g_auto/f_auto/q_auto/v1/author/jeremy-mccarter?_a=BAVARSDW0 96w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_128,h_128,g_auto/f_auto/q_auto/v1/author/jeremy-mccarter?_a=BAVARSDW0 128w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_256,h_256,g_auto/f_auto/q_auto/v1/author/jeremy-mccarter?_a=BAVARSDW0 256w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_384,h_384,g_auto/f_auto/q_auto/v1/author/jeremy-mccarter?_a=BAVARSDW0 384w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_640,h_640,g_auto/f_auto/q_auto/v1/author/jeremy-mccarter?_a=BAVARSDW0 640w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_750,h_750,g_auto/f_auto/q_auto/v1/author/jeremy-mccarter?_a=BAVARSDW0 750w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_828,h_828,g_auto/f_auto/q_auto/v1/author/jeremy-mccarter?_a=BAVARSDW0 828w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_1080,h_1080,g_auto/f_auto/q_auto/v1/author/jeremy-mccarter?_a=BAVARSDW0 1080w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_1200,h_1200,g_auto/f_auto/q_auto/v1/author/jeremy-mccarter?_a=BAVARSDW0 1200w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_1920,h_1920,g_auto/f_auto/q_auto/v1/author/jeremy-mccarter?_a=BAVARSDW0 1920w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_2048,h_2048,g_auto/f_auto/q_auto/v1/author/jeremy-mccarter?_a=BAVARSDW0 2048w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_3840,h_3840,g_auto/f_auto/q_auto/v1/author/jeremy-mccarter?_a=BAVARSDW0 3840w"
                style="color: transparent;"
                width="207"
              />
            </figure>
          </a>
          <p
            class="font-primary text-pretty text-2xl line-clamp-2 mt-2"
          >
            <a
              href="/author/jeremy-mccarter"
              rel="author"
              title="Jeremy McCarter"
            >
              Jeremy McCarter
            </a>
          </p>
        </article>
        <article
          class="p-5 w-[227px] h-[267px] flex-none text-sm text-pretty"
        >
          <a
            href="/author/linmanuel-miranda"
            rel="author"
            title="Lin-Manuel Miranda"
          >
            <figure
              class="bg-green-300 w-[187px] h-[187px] relative overflow-hidden rounded-full"
            >
              <img
                alt="Lin-Manuel Miranda"
                class="transition-all w-[187px] h-[187px] hover:w-[207px] hover:h-[207px] hover:-mt-[10px] object-cover"
                data-nimg="1"
                decoding="async"
                height="207"
                loading="lazy"
                sizes="207px"
                src="https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_3840,h_3840,g_auto/f_auto/q_auto/v1/author/linmanuel-miranda?_a=BAVARSDW0"
                srcset="https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_16,h_16,g_auto/f_auto/q_auto/v1/author/linmanuel-miranda?_a=BAVARSDW0 16w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_32,h_32,g_auto/f_auto/q_auto/v1/author/linmanuel-miranda?_a=BAVARSDW0 32w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_48,h_48,g_auto/f_auto/q_auto/v1/author/linmanuel-miranda?_a=BAVARSDW0 48w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_64,h_64,g_auto/f_auto/q_auto/v1/author/linmanuel-miranda?_a=BAVARSDW0 64w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_96,h_96,g_auto/f_auto/q_auto/v1/author/linmanuel-miranda?_a=BAVARSDW0 96w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_128,h_128,g_auto/f_auto/q_auto/v1/author/linmanuel-miranda?_a=BAVARSDW0 128w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_256,h_256,g_auto/f_auto/q_auto/v1/author/linmanuel-miranda?_a=BAVARSDW0 256w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_384,h_384,g_auto/f_auto/q_auto/v1/author/linmanuel-miranda?_a=BAVARSDW0 384w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_640,h_640,g_auto/f_auto/q_auto/v1/author/linmanuel-miranda?_a=BAVARSDW0 640w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_750,h_750,g_auto/f_auto/q_auto/v1/author/linmanuel-miranda?_a=BAVARSDW0 750w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_828,h_828,g_auto/f_auto/q_auto/v1/author/linmanuel-miranda?_a=BAVARSDW0 828w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_1080,h_1080,g_auto/f_auto/q_auto/v1/author/linmanuel-miranda?_a=BAVARSDW0 1080w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_1200,h_1200,g_auto/f_auto/q_auto/v1/author/linmanuel-miranda?_a=BAVARSDW0 1200w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_1920,h_1920,g_auto/f_auto/q_auto/v1/author/linmanuel-miranda?_a=BAVARSDW0 1920w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_2048,h_2048,g_auto/f_auto/q_auto/v1/author/linmanuel-miranda?_a=BAVARSDW0 2048w, https://res.cloudinary.com/dulz2k2my/image/upload/c_auto,w_3840,h_3840,g_auto/f_auto/q_auto/v1/author/linmanuel-miranda?_a=BAVARSDW0 3840w"
                style="color: transparent;"
                width="207"
              />
            </figure>
          </a>
          <p
            class="font-primary text-pretty text-2xl line-clamp-2 mt-2"
          >
            <a
              href="/author/linmanuel-miranda"
              rel="author"
              title="Lin-Manuel Miranda"
            >
              Lin-Manuel Miranda
            </a>
          </p>
        </article>
      </div>
    </section>
  </div>
</div>
`;
